spec/classes/python_spec.rb

```diff
@@ -271,4 +271,101 @@
       end
     end
   end
+
+  context "on FreeBSD OS" do
+    let :facts do
+      {
+          :id                     => 'root',
+          :kernel                 => 'FreeBSD',
+          :osfamily               => 'FreeBSD',
+          :operatingsystem        => 'FreeBSD',
+          :operatingsystemrelease => '10.2-RELEASE',
+          :path                   => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
+          :concat_basedir         => '/dne',
+      }
+    end
+
+    it { is_expected.to contain_class("python::install") }
+    # Base debian packages.
+    it { is_expected.to contain_package("python") }
+    it { is_expected.to_not contain_package("python-dev") }
+    # Python 2.7 system packages
+    it { is_expected.to contain_package("py27-virtualenv")}
+
+    context "with python 2" do
+      context "with pip" do
+        let(:params) {{:version => 'system', :pip => true}}
+        it { is_expected.to contain_package("py27-pip") }
+      end
+    end
+
+    context "with python 3" do
+      let (:params) {{:version => '3'}}
+      it { is_expected.to contain_package('python3')}
+
+      context "with pip" do
+        let (:params) {{:version => '3', :pip => true}}
+        it { is_expected.to contain_exec('install_pip34')}
```

Comment:
The resources in the spec tests should have their attributes listed along with their values, though given that the existing code is not doing this, I think it is OK.